File: Api/Api.Domain/Entities/Course.cs
﻿using System;
using System.Collections.Generic;

namespace Api.Domain.Entities
{
    /// <summary>
    /// Course entity
    /// </summary>
    public class Course : Schedule
    {
        /// <summary>
        /// Get or Set Course Key
        /// </summary>
        public Guid CourseKey { get; set; }

        /// <summary>
        /// Get or Set Course Name
        /// </summary>
        public string CourseName { get; set; }

        /// <summary>
        /// Get or Set Course Type
        /// </summary>
        public Guid Type { get; set; }

        /// <summary>
        /// Get or Set Course Units
        /// </summary>
        public IEnumerable<CourseUnit> CourseUnits { get; set; }
    }
}

File: App/src/app/models/entities/Course.ts
import { Guid } from 'guid-typescript';
import { CourseUnit } from './CourseUnit';

export interface Course {
  scheduleKey: Guid,
  scheduleVersion: number,
  courseKey: Guid,
  courseName: string,
  type: number,
  courseUnits: CourseUnit[]
}

